mojaloop/pisp-demo-server
# Setup API endpoints

## Summary

The following are the changes made in this PR:
- Add Open API definitions for the PISP demo server to interact with the mobile app and Mojaloop
- Split the communication for mojaloop and app into virtual hosts
- Add bare-bones handlers 
- Fix tests and various checks to pass circle CI

Note:
- Changed the `audit:resolve` and `audit:check` scripts to only evaluate production dependencies. Otherwise, we will need a `audit-resolve.json` file of ~42k lines to ignore the checks related to the dev dependencies (`lodash`, `commitlint`, etc.). 
- Currently the minimum coverage for unit tests is removed (changed from `80` to `10`) since the PR is already looking too huge. I will add it using other PRs as the handlers are being implemented.

## APIs for [pisp-demo-app-flutter](https://github.com/mojaloop/pisp-demo-app-flutter)

How to check:
- All of the path definitions are written in `src/interface/app.yaml`
- To view this, execute `npm run build:openapi:app`. The bundled Open API file will be written on `dist/openapi/app.yaml`
- Copy and paste the bundled file to a swagger editor

TODO (in other PR):
- Serve the swagger documentation so that people don't need to copy-paste the definition to a swagger editor.
- Implement handlers logic
- Add API definitions for polling by the mobile app

## APIs for communication with Mojaloop

How to check:
- All of the path definitions are written in `src/interface/mojaloop.yaml`
- To view this, execute `npm run build:openapi:mojaloop`. The bundled Open API file will be written on `dist/openapi/mojaloop.yaml`
- Copy and paste the bundled file to a swagger editor

TODO (in other PR):
- Serve the swagger documentation so that people don't need to copy-paste the definition to a swagger editor.
- Implement handlers logic
- Confirm the models with what will be used by the third-party adapter.
